.github/workflows/build-test-pr.yml

```diff
@@ -0,0 +1,51 @@
+name: Gradle Build and Test Pull Requests to OpenHouse
```

Comment:
Could you help me understand why this needs to be separated in a different workflow? Is it because of the latency of the work queue for the build-tag-publish workflow?

Reply:
The main motivation is a better separation of concerns.

The original workflow's primary purpose is to test the branch, build the images and push them to docker and JFrog. However, in PRs it would skip the docker image generation and publishing step since that would only apply on the main branch.

My reasoning for splitting it up is that the logic of the two workflows seem to diverge quite a bit. For checkout on the main branch its recommended to use the merge commit hash I believe in scenarios where the commits are not squashed-then-merged (e.g. a rebase merge I believe) so that the merge commit is consistent for a single PR.

It's possible to write an if/else logic in the workflow and reuse them for both PRs and main branch but I find there are other benefits of splitting them up to, such as defining a concurrency policy that makes sense for PRs which would not make sense for a main branch push.

Reply:
Thanks for the clarification, if the intention is the separate these out while they are still using mostly the same steps, lets modularize these actions (see https://devblogs.microsoft.com/ise/three-ways-to-simplify-cicd-pipelines-on-github-actions/) that way when we change how we want to build/test we only have to change it in a single place, while the build-tag-publish could leverage the testing flow and finish with artifactory publish steps

Reply:
Good catch I'll try that